pyfar/classes/coordinates.py

```diff
@@ -1374,16 +1374,347 @@ def show(self, mask=None, **kwargs):
+
+        Notes
+        -----
+        This is a wrapper for ``scipy.spatial.cKDTree``. Other than previous
+        implementations, it supports self.ndim>1 as well.
+
+        Examples
+        --------
+
+        Find frontal point from a spherical coordinate system
+
+        .. plot::
+
+            >>> import pyfar as pf
+            >>> import numpy as np
+            >>> coords = pf.samplings.sph_lebedev(sh_order=10)
+            >>> find = pf.Coordinates(1, 0, 0)
+            >>> index, distance = coords.find_nearest(find)
+            >>> mask = np.zeros(coords.cshape, dtype=bool)
+            >>> mask[index] = True
+            >>> coords.show(mask)
+            >>> distance
+            >>> 0.0
+
+        Find multidimensional points in multidimensional coordinates with k=1
+
+        >>> import pyfar as pf
+        >>> import numpy as np
+        >>> coords = pf.Coordinates(np.arange(9).reshape((3, 3)), 0, 1)
+        >>> find = pf.Coordinates(
+        >>>     np.array([[0, 1], [2, 3]]), 0, 1)
+        >>> i, d = coords.find_nearest(find)
+        >>> coords[i] == find
+        True
+        >>> i
+        (array([[0, 0],
+                [0, 1]], dtype=int64),
+         array([[0, 1],
+                [2, 0]], dtype=int64))
+        >>> d
+        array([[0., 0.],
+               [0., 0.]])
+
+        Find multidimensional points in multidimensional coordinates with k=3
+
+        >>> import pyfar as pf
+        >>> import numpy as np
+        >>> coords = pf.Coordinates(np.arange(9).reshape((3, 3)), 0, 1)
+        >>> find = pf.Coordinates(
+        >>>     np.array([[0, 1], [2, 3]]), 0, 1)
+        >>> i, d = coords.find_nearest(find, 3)
+        >>> # the k-th dimension is at the end
+        >>> i[0].shape
+        (3, 2, 2)
+        >>> # now just access the k=0 dimension
+        >>> coords[i][0].cartesian
+        array([[[0., 0., 1.],
+                [1., 0., 1.]],
+               [[2., 0., 1.],
+                [3., 0., 1.]]])
+        """
+
+        # check the input
+        if not isinstance(k, int) or k <= 0 or k > self.csize:
+            raise ValueError("k must be an integer > 0 and <= self.csize.")
+        if not isinstance(coords, Coordinates):
+            raise ValueError("coords must be an pf.Coordinates object.")
+        if distance_measure not in ['euclidean', 'spherical']:
+            raise ValueError(
+                "distance_measure needs to be 'euclidean' or 'spherical' and "
+                f"it is {distance_measure}")
+
+        # get target point in cartesian coordinates
+        points = coords.cartesian
+
+        # get KDTree
+        kdtree = self._make_kdtree()
+
+        # query nearest neighbors
+        points = points.flatten() if coords.csize == 1 else points
+
+        # nearest points
+        distance, index = kdtree.query(points, k=k)
+
+        if distance_measure == 'spherical':
+            # determine validate radius
+            radius = self.radius
```

Comment:
I think we should use `np.concatenate((self.radius, coords.radius))` for `radius`